File: Dataset/Leetcode/train/102/705.js
 var XXX = function(root) {
    let answer = []
    re = function (root, curr){
        if(!root) return
        if(answer[curr]) {//check cache
            answer[curr].push(root.val)
        }else{//create
            answer[curr] = [root.val]
        }
        if(root.left){re(root.left,curr + 1)}
        if(root.right){re(root.right,curr + 1)}
    }
    
    re(root,0)
    return answer    
};

File: Dataset/Leetcode/train/94/705.cpp
 class Solution {
public:
    vector<int> XXX(TreeNode* root) {
        vector<int> ans;
        stack<TreeNode*>    sta;
        TreeNode* p;
        for(p = root ; p ; p = p -> left)
            sta.push(p);
        while(!sta.empty()){
            p = sta.top();
            ans.push_back(p -> val);
            sta.pop();
            for(p = p -> right ; p ; p = p -> left)
                sta.push(p);
        }
        return  ans;
    }
};

